source/common/network/listen_socket_impl.h

```diff
@@ -116,7 +116,8 @@ class ConnectionSocketImpl : public SocketImpl, public ConnectionSocket {
   }
 
   void setRequestedServerName(absl::string_view server_name) override {
-    server_name_ = std::string(server_name);
+    // Always keep the server_name_ as lower case.
+    server_name_ = absl::AsciiStrToLower(std::string(server_name));
```

Comment:
I think the overload of `absl::AsciiStrToLower` called here takes a `string_view` not a `std::string`, so the `std::string` conversion here should be removed.

Reply:
got it, will update